liberty-maven-plugin/src/main/java/io/openliberty/tools/maven/server/PluginConfigSupport.java

```diff
@@ -305,7 +305,7 @@ protected boolean isAnyAppConfiguredInSourceServerXml() {
 
         File serverXML = new File(serverDirectory, "server.xml");
 
-        if (serverXML != null && serverXML.exists()) {
+        if (isConfigCopied() && serverXML != null && serverXML.exists()) {
```

Comment:
Won't this cause some scenario to fail since it is not returning the app config locations for the source server xml if the config files have not been copied yet? Can we just go ahead and call copyConfigFiles() here if isConfigCopied() returns false?

Reply:
I don't think that would work with every case. The main problem here was that we created and stored the SCD when we generate the liberty-plugin-config.xml as part of an install with installServerAssembly(). Part of that is exportParametersToXml() which calls the problem method getAppConfigLocationsFromSourceServerXml(). In that case we wouldn't be able to copy the config since no server was created.

Reply:
I think the only situation that this change eliminates is the initial plugin-config.xml generation that would get overwritten as part of Deploy once the config is loaded.

Reply:
Would moving the check out into exportParametersToXml() would be a better solution? In that case we wouldn't be introducing this behavior into any other scenarios I didn't run through. We would just be skipping the call in one spot rather than any other parts of the plugin that call getAppConfigLocationsFromSourceServerXml().

Reply:
I like the idea of doing the check in exportParametersToXml(). So did you mean that you would check isConfigCopied() at the beginning of the method and skip the entire method body if that returns false?

Reply:
Just the part that calls getAppsDirectory(). We would still call the rest of the method.

Reply:
It seems like some of the rest of that method doesn't make sense either (saving off properties and file locations that have not been copied over yet either). Does all of that info need to be written out to this file when it is not even valid?